src/typescriptGenerator.ts

```diff
@@ -87,12 +71,33 @@ export async function generateTypeScriptLibrary(
   ))
     ? true
     : armLibrary;
+  const useCoreV2: boolean = (await host.GetValue("use-core-v2")) || false;

```

Comment:
Can we make this true by default?

My worry about having this false by default is that even when we get these changes in, people will keep creating new SDKs using core v1, and we'll need to do the migrations at one point.

I think it is better to force core v2 as soon as this is merged and set `use-core-v2: false` for spceial cases

Reply:
As mentioned in one of the comments by Deya, below, we are going to turn it on by default once all packages have moved to core V2. This is to support the patch version releases.

Reply:
I am changing the code so the default mode will be use-core-v2:true. SDKs who want core V1 can set it to false.